### ISSUE TITLE
Provide timeout for Github Actions

### ISSUE THREAD

**Issue by Bukama** [junit-pioneer/junit-pioneer#322]
Currently sometimes our builds run "forever" and are killed after six hours, e.g. [run 988295906](https://github.com/junit-pioneer/junit-pioneer/pull/321/checks?check_run_id=988295906). 

When I remember correctly there was a timeout flag you can provide to the actions to timeout after a certain of time, maybe 15mins or something. 

Am I right (@aepfli @beatngu13 ) and do we want such a thing?

**Comment by beatngu13** [comment]
Thanks for making me aware of this. The example job you posted is with Java 11 on Windows, and it looks like it timed out during the tests:

```none
2020-08-15T13:32:27.2087572Z > Task :processTestResources
2020-08-15T13:32:27.2087905Z > Task :testClasses
2020-08-15T13:32:57.2085938Z > Task :test
2020-08-15T19:30:51.8898754Z ##[error]The operation was canceled.
2020-08-15T19:30:51.9603377Z Post job cleanup.
2020-08-15T19:30:52.1317368Z Post job cleanup.
2020-08-15T19:30:54.1273868Z [command]"C:\Program Files\Git\bin\git.exe" version
```

Would be interesting to see if this always happens on the same job during the same Gradle task. Do you have more examples @Bukama?

In the meantime, I will open PR do include an appropriate timeout.

**Comment by Bukama** [comment]
> Thanks for making me aware of this. The example job you posted is with Java 11 on Windows, and it looks like it timed out during the tests:
>
> [...]}

> Would be interesting to see if this always happens on the same job during the same Gradle task. Do you have more examples @Bukama?

[961045944 - win / java 11](https://github.com/junit-pioneer/junit-pioneer/pull/261/checks?check_run_id=961045944)

I don't remember where, but there were one or two more runs, but as I don't remember the runs I don't know if they were also win / java 11

**Comment by aepfli** [comment]
https://github.com/junit-pioneer/junit-pioneer/runs/990846534

**Comment by beatngu13** [comment]
So this time Java 14, but still on Windows.

**Comment by aepfli** [comment]
should we increase the logoutput to info for a while, so we have more information?

**Comment by beatngu13** [comment]
Sounds reasonable to me. But before doing so, we should probably create a separate issue (will take care) for the build timeout since this here will be closed once #323 is merged. Then we can collect all necessary information and decisions there.

**Comment by Bukama** [comment]
I merged #323 to have the timeout in the main build.

According to your question: I think timeouts do not hurt in all jobs. @aepfli Do you thing 15mins is enough for the deployment to bintray/nexus jobs?

I'll leave the issue open to add timeouts to the other jobs.
